src/main/java/com/monumental/services/UserService.java

```diff
@@ -58,23 +59,41 @@ public User signup(CreateUserRequest userRequest, String appUrl, Locale locale)
         user.setIsEmailVerified(false);
         this.userRepository.save(user);
 
-        this.sendVerificationEmail(user, this.generateVerificationToken(user), appUrl, locale);
+        this.sendVerificationEmail(user, this.generateVerificationToken(user, VerificationToken.Type.SIGNUP), appUrl, locale);
 
         return user;
     }
 
-    public VerificationToken generateVerificationToken(User user) {
+    public void resetPassword(String email, String appUrl, Locale locale) {
+        User user = this.userRepository.getByEmail(email);
+        // Note: This is a security feature. We don't want the password reset form to tell everyone what email addresses are registered
+        if (user == null) {
+            return;
+        }
+        this.sendPasswordResetEmail(user, this.generateVerificationToken(user, VerificationToken.Type.PASSWORD_RESET), appUrl, locale);
+    }
+
+    public void deleteAllVerificationTokensOfTypeForUser(User user, VerificationToken.Type type) {
+        List<VerificationToken> tokens = this.tokenRepository.findAllByUserAndType(user, type);
+        if (tokens != null && tokens.size() > 0) {
+            this.tokenRepository.deleteAll(tokens);
+        }
+    }
```

Comment:
Couldn't we just use something like `this.tokenRepository.deleteAllByUserAndType(user, type);`?